uisdk/src/main/java/com/karhoo/uisdk/screen/booking/BookingActivity.kt

```diff
@@ -271,9 +276,9 @@ class BookingActivity : BaseActivity(), AddressBarMVP.Actions, BookingMapMVP.Act
     }
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        if (requestCode == REQ_CODE_BRAINTREE || requestCode == REQ_CODE_BRAINTREE_GUEST ||
-                requestCode == REQ_CODE_ADYEN) {
-            bookingRequestWidget.onActivityResult(requestCode, resultCode, data)
+        if(resultCode == Activity.RESULT_OK && requestCode == REQ_CODE_BOOKING_REQUEST_ACTIVITY) {
+            waitForTripAllocation()
+            tripAllocationWidget.waitForAllocation(data?.extras?.get(BookingRequestActivity.BOOKING_REQUEST_TRIP_INFO_KEY) as TripInfo)
```

Comment:
It is not a good idea to directly call a public method on the widget. Perhaps it is better to create a contact between tripAllocationWidget and outside (using an interface ) similar to
```
interface AddressBarViewContract {
    interface Widget {
                 fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?)
   }
}
```

This way we create an abstraction between the views/widgets and their usages.

Reply:
I agree, not sure if we should do it during this ticket though since this logic was already there, it's old and could use a bit of refactoring.
The other widgets are called in the same way also:

`override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (requestCode == REQ_CODE_BRAINTREE || requestCode == REQ_CODE_BRAINTREE_GUEST ||
                requestCode == REQ_CODE_ADYEN) {
            bookingRequestWidget.onActivityResult(requestCode, resultCode, data)
        } else if (resultCode == RESULT_OK && data != null) {
            when (requestCode) {
                AddressCodes.PICKUP -> addressBarWidget.onActivityResult(requestCode, resultCode, data)
                AddressCodes.DESTINATION -> addressBarWidget.onActivityResult(requestCode, resultCode, data)
            }
        }
        super.onActivityResult(requestCode, resultCode, data)
    }`

Reply:
We call a public method on them but the method is part of the `interface AddressBarViewContract`

what I would suggest doing is to define an interface like 

```
interface TripAllocationMVP { /// I would not mind renaming it to TripAllocationContract
       interface Widget {
          fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?)
       }
}
```

then 
```
class TripAllocationView .....: TripAllocationMVP.Widget
{
override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
       /// handle the response
    }
```

and in the BookingActivity
```
    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if(resultCode == Activity.RESULT_OK && requestCode == REQ_CODE_BOOKING_REQUEST_ACTIVITY) {
            waitForTripAllocation()
            tripAllocationWidget.onActivityResult(requestCode, resultCode, data)
   }
}
```
This way the tripAllocationWidget would have the logic of handling the onActivityResult